openedx/features/course_experience/tests/views/test_course_home.py

```diff
@@ -219,32 +219,44 @@ def tearDown(self):
-        [CourseUserType.UNENROLLED_STAFF, 'You must be enrolled in the course to see course content.'],
+        [False, 'private', CourseUserType.ANONYMOUS, 'You must be enrolled in the course to see course content.'],
+        [False, 'preview', CourseUserType.ANONYMOUS, 'You must be enrolled in the course to see course content.'],
+        [False, 'public', CourseUserType.ANONYMOUS, 'You must be enrolled in the course to see course content.'],
+        [True, 'private', CourseUserType.ANONYMOUS, 'You must be enrolled in the course to see course content.'],
+        [True, 'preview', CourseUserType.ANONYMOUS, 'You must be enrolled in the course to see course content.'],
+        [True, 'public', CourseUserType.ANONYMOUS, 'You must be enrolled in the course to see course content.'],
+        [False, 'private', CourseUserType.ENROLLED, None],
+        [False, 'private', CourseUserType.UNENROLLED, 'You must be enrolled in the course to see course content.'],
+        [False, 'private', CourseUserType.UNENROLLED_STAFF,
+         'You must be enrolled in the course to see course content.'],
     )
     @ddt.unpack
-    def test_home_page(self, user_type, expected_message):
+    def test_home_page(self, enable_anonymous_access, course_visibility, user_type, expected_message):
         self.create_user_for_course(self.course, user_type)
 
         # Render the course home page
-        url = course_home_url(self.course)
-        response = self.client.get(url)
+        with mock.patch('xmodule.course_module.CourseDescriptor.course_visibility', course_visibility):
+            # Test access with anonymous flag and course visibility
+            with override_waffle_flag(COURSE_ENABLE_ANONYMOUS_ACCESS_FLAG, enable_anonymous_access):
+                url = course_home_url(self.course)
+                response = self.client.get(url)
 
         # Verify that the course tools and dates are always shown
         self.assertContains(response, 'Course Tools')
         self.assertContains(response, 'Today is')
 
-        # Verify that the outline, start button, course sock, and welcome message
+        # Verify that start button, course sock, and welcome message
         # are only shown to enrolled users.
         is_enrolled = user_type is CourseUserType.ENROLLED
         is_unenrolled_staff = user_type is CourseUserType.UNENROLLED_STAFF
-        expected_count = 1 if (is_enrolled or is_unenrolled_staff) else 0
-        self.assertContains(response, TEST_CHAPTER_NAME, count=expected_count)
-        self.assertContains(response, 'Start Course', count=expected_count)
+        is_public = course_visibility == 'public' and enable_anonymous_access
+        is_preview = course_visibility == 'preview' and enable_anonymous_access
+        expected_welcome = 1 if (is_enrolled or is_unenrolled_staff) else 0
+        expected_chapter = 1 if (is_public or is_preview) else expected_welcome
```

Comment:
While the checks are all correct, they can be made clearer to read by moving `self.assertContains(response, TEST_CHAPTER_NAME, count=expected_chapter_count)` and all the variables that are only needed for it to line 261 (and leave an empty line). You can add a comment like "# Verify the outline is shown for enrolled users and if needed for anonymous users."

Reply:
Also is your editor line width set to 80? edX uses 120 and I think there are a lot of multi-lines in this PR which would not be.

Reply:
Yeah, that makes it more readable.

On the multilines, I've must have set that manually for readability, I'll review it and update accordingly.